### PR TITLE
Sean Naleid: webapi-challenge

### PR DESCRIPTION
**Pull request by seanaleid**

### CODE REVIEW

**Comment by seanaleid** [comment]
Deployed API on Heroku at: https://webapi-projects-challenge.herokuapp.com/
(although continues getting the 503 error from yesterday)
Created and deployed react-app at: webapiprojects.now.sh

File: README.md [hunk 1]
```diff
@@ -26,29 +27,39 @@ Demonstrate your understanding of this Sprint's concepts by answering the follow
 
 - [ ] Mention two parts of Express that you learned about this week.
 
+    Two parts of express we learned this week are server side routing and middleware.
```
Comment (MrT3313) [inline README.md:30]: can and what do these do within express?

File: README.md [hunk 2]
```diff
@@ -26,29 +27,39 @@ Demonstrate your understanding of this Sprint's concepts by answering the follow
 
 - [ ] Mention two parts of Express that you learned about this week.
 
+    Two parts of express we learned this week are server side routing and middleware. 
+
 - [ ] Describe Middleware?
 
+    Middleware is just a function that we put between the path and the homies ((req, res)) inside a back-end route. Middleware adds something or validates the code being passed through it.
+
```
Comment (MrT3313) [inline README.md:35]: 👍 
edit: middleware adds ANY functionality you can code into a function!

File: README.md [hunk 3]
```diff
@@ -26,29 +27,39 @@ Demonstrate your understanding of this Sprint's concepts by answering the follow
...
 - [ ] Describe a Resource?
 
+    A resource is basically data, it's whatever we put on the server.
```
Comment (MrT3313) [inline README.md:38]: category / grouping of data

File: README.md [hunk 4]
```diff
@@ -26,29 +27,39 @@ Demonstrate your understanding of this Sprint's concepts by answering the follow
...
 
+    Middleware is just a function that we put between the path and the homies ((req, res)) inside a back-end route. Middleware adds something or validates the code being passed through it.
+
 - [ ] Describe a Resource?
 
+    A resource is basically data, it's whatever we put on the server. 
+
 - [ ] What can the API return to help clients know if a request was successful?
 
+    A status code (100s, 200s, 300s, 400s, or 500s). A successfully completed task returns a status code in the 200s.
```
Comment (MrT3313) [inline README.md:42]: these are the possible ranges but specifically for 'success' responses, it is the 200 range

File: README.md [hunk 5]
```diff
@@ -26,29 +27,39 @@ Demonstrate your understanding of this Sprint's concepts by answering the follow
...
 
+    A resource is basically data, it's whatever we put on the server. 
+
 - [ ] What can the API return to help clients know if a request was successful?
 
+    A status code (100s, 200s, 300s, 400s, or 500s). A successfully completed task returns a status code in the 200s. 
+
 - [ ] How can we partition our application into sub-applications?
 
+    We can partition our app into sub-apps by using express routing and connecting everything properly.
```
Comment (MrT3313) [inline README.md:46]: 👍

File: index.js [hunk 6]
```diff
@@ -1,3 +1,14 @@
+if(!process.env.now) require('dotenv').config();
```
Comment (MrT3313) [inline index.js:1]: I don't think this is doing what you think it is doing. an if statement needs to be contained in a scope

`if ( checkThisCondition ) {if TRUE -> go in this scope}`

File: index.js [hunk 7]
```diff
@@ -1,3 +1,14 @@
...
+
+const server = require("./server.js");
+
+const port = process.env.now ? 8080 : 4000;
```
Comment (MrT3313) [inline index.js:5]: .now deployment doesn't automatically set its port? it allows you to set the port yourself even when it is building?

File: data/helpers/projectRouter.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
+// imports express and creates the router (chaining them together)
+const router = require('express').Router();
+
+// import model
+const projectDb = require('./projectModel.js');
+
+
+var express = require('express')
+var cors = require('cors')
```
Comment (MrT3313) [inline data/helpers/projectRouter.js:9]: you can add cors on the server.js so you dont need to add it to each router

**Review by MrT3313 — APPROVED** [review]
Really good job this week! Strong submission and our best 1:1 as far as confidence in your code goes!

🎉 🎉

**Pull request merged**